brooklyn-tosca-common/src/main/java/io/cloudsoft/tosca/metadata/BrooklynToscaTypeProvider.java

```diff
@@ -1,20 +1,47 @@
 package io.cloudsoft.tosca.metadata;
 
-import java.util.Map;
-
 import com.google.common.base.Optional;
 import com.google.common.collect.ImmutableMap;
+import org.apache.brooklyn.rest.client.BrooklynApi;
+import org.apache.brooklyn.rest.domain.CatalogEntitySummary;
 
-public class BrooklynToscaTypeProvider implements ToscaTypeProvider {
+import java.util.List;
+import java.util.Map;
+
+public class BrooklynToscaTypeProvider implements ToscaTypeProvider, RequiresBrooklynApi {
 
     private Map<String, String> typeMapping = ImmutableMap.<String, String>builder()
+            .put("org.apache.brooklyn.entity.proxy.LoadBalancer", "brooklyn.nodes.LoadBalancer")
+            .put("org.apache.brooklyn.entity.database.DatabaseNode", "brooklyn.nodes.Database")
             .put("org.apache.brooklyn.entity.database.mysql.MySqlNode", "brooklyn.nodes.Database")
-            .put("org.apache.brooklyn.entity.webapp.tomcat.TomcatServer", "brooklyn.nodes.WebServer")
-            .put("org.apache.brooklyn.entity.webapp.tomcat.Tomcat8Server", "brooklyn.nodes.WebServer")
+            .put("org.apache.brooklyn.entity.webapp.WebAppService", "brooklyn.nodes.WebServer")
             .build();
 
+    private BrooklynApi brooklynApi;
+
     @Override
-    public Optional<String> getToscaType(String type) {
+    public void setBrooklynApi(BrooklynApi brooklynApi) {
+        this.brooklynApi = brooklynApi;
+    }
+
+    @Override
+    public Optional<String> getToscaType(String type, String version) {
+        try {
+            CatalogEntitySummary catalogEntitySummary = brooklynApi.getCatalogApi().getEntity(type, version);
+            for (Object o : catalogEntitySummary.getTags()) {
+                Map<String, ?> map = (Map<String, ?>) o;
+                if (map.containsKey("traits")) {
+                    List<String> traits = (List<String>) map.get("traits");
```

Comment:
Create a set instead? Though expect collection to be small enough that it's negligible.